### DISCUSSION TITLE
chat record

### DISCUSSION THREAD

**Discussion by B10610204** [chaskiq/chaskiq#563] [Q&A] [answered]
Is there any detailed function introduction?
How to save conversation records?

**Comment by michelson** [comment, marked as answer]
Hi, there is no information, but you could check app/spec/models/conversation_spec.rb . The API is relatively simple though

```ruby
app.start_conversation(
      message: { text_content: "aa" },
      from: agent_role.agent,
      participant: app_user
    )
```

the message could be `message: { text_content: "aa" },` or `message: { serialized_content: {....} },` or message: {blocks: {...}}


serialized_content is the Chaskiq text editor. the blocks are the API plugin UI in wich you can render content from plugins
